.github/workflows/_rust.yml

```diff
@@ -119,3 +119,26 @@ jobs:
       - name: Run smoke tests (Windows)
         if: ${{ runner.os == 'Windows' }}
         run: bash ../../scripts/tests/smoke-test-gui-windows.sh
+
+  headless-client:
+    name: headless-client-${{ matrix.runs-on }}
```

Comment:
I named this `headless-client`, similar to `gui-client-smoke-test`, figuring that we may add other tests in here that need a binary, a real kernel, and systemd, (i.e. can't run in Docker easily) but which don't actually need the portal running (i.e. aren't integration tests)